src/features/metadata-based-view/IconWithTooltip.js

```diff
@@ -0,0 +1,51 @@
+// @flow strict
+
+import React, { type Element } from 'react';
+import { injectIntl } from 'react-intl';
+
+import PlainButton from '../../components/plain-button';
+import Tooltip from '../../components/tooltip';
+import IconCheck from '../../icons/general/IconCheck';
+import IconClose from '../../icons/general/IconClose';
+import IconPencil from '../../icons/general/IconPencil';
+
+import { bdlGray62 } from '../../styles/variables';
+
+import { CANCEL_ICON_TYPE, EDIT_ICON_TYPE, SAVE_ICON_TYPE } from './constants';
+
+type IconType = typeof CANCEL_ICON_TYPE | typeof EDIT_ICON_TYPE | typeof SAVE_ICON_TYPE;
+type Props = {
+    className?: string,
+    onClick: void => void,
+    tooltipText: String,
+    type?: IconType,
+} & InjectIntlProvidedProps;
+
+const IconWithTooltip = ({ className, onClick, tooltipText, type }: Props): Element<typeof Tooltip> | null => {
+    let icon;
+
+    switch (type) {
+        case CANCEL_ICON_TYPE:
+            icon = <IconClose color={bdlGray62} width={16} height={16} />;
+            break;
+        case EDIT_ICON_TYPE:
+            icon = <IconPencil />;
```

Comment:
Does the pencil icon not need the color and dimensions specified?

Reply:
The default props for size and color look fine for this use case.